Bundle/QueryBundle/Helper/QueryHelper.php

```diff
@@ -185,6 +190,14 @@ public function buildWithSubQuery($containerEntity, QueryBuilder $itemsQueryBuil
             }
         }
 
+        if (strpos($query, ':currentUser') !== false && is_object($this->getCurrentUser())) {
+            if (is_object($this->getCurrentUser())) {
+                $itemsQueryBuilder->setParameter('currentUser', $this->getCurrentUser()->getId());
+            } else {
+                throw new AccessDeniedException();
```

Comment:
I wasn't able to find a case where we would like to have such a widget in an anonymous scope but why not.

To go further, I think we could just do not run the query if there is the :currentUser parameter and no user authenticated